add support for Tables

bardurdam opened the pull request:
- add `Table`, `TableColumn`, `TableStyle` and `TableStyleName` to
  model.

- make `Model2XlsxConversions` build tables from model.

- add example for defining a table

- tell Travis to use JDK8

norbert-radyk commented on src/main/scala/com/norbitltd/spoiwo/model/Table.scala:
```diff
@@ -0,0 +1,33 @@
+    enableAutoFilter: java.lang.Boolean = null
+  ): Table =
+    Table(
+      cellRange,
```
Please use named parameters as in i.e. `Sheet`. They do add a bit code in a short run, but are much safer especially with many attributes of the same type (i.e. you will not mix up name and display name by accident).

norbert-radyk commented on src/main/scala/com/norbitltd/spoiwo/natures/xlsx/Model2XlsxConversions.scala:
```diff
@@ -274,9 +274,64 @@ object Model2XlsxConversions {
     sheet
   }
 
+  private[xlsx] def convertTable(modelTable: Table, sheet: XSSFSheet): Unit = {
```
Where is the Spec for this conversion?

norbert-radyk commented on src/main/scala/com/norbitltd/spoiwo/natures/xlsx/Model2XlsxConversions.scala:
```diff
@@ -274,9 +274,64 @@ object Model2XlsxConversions {
+
+
+    // Columns
+    val modelColumns = if(modelTable.columns.isEmpty) {
```
So you are checking here if the `modelTable.columns` are empty and then fill with the number of columns from start to end. What will happen if `modelTable.columns.size < (eCol - sCol) + 1`

bardurdam commented:
@norbert-radyk, thanks for the feedback.

I have updated the PR.

I have set a validation wrt. table columns, so if they are specified, but do not equal the amount of needed columns, you will get a runtime exception. Table columns are therefore only generated when not speficied at all.

I've added specs too.
